File: test-suite/src/test/groovy/io/micronaut/docs/server/uris/UriTemplateTest.java
package io.micronaut.docs.server.uris;

import io.micronaut.http.uri.UriMatchTemplate;
import static org.junit.Assert.*;
import org.junit.Test;

import java.util.Collections;

public class UriTemplateTest {

    @Test
    public void testUriTemplate() {

        // tag::match[]
        UriMatchTemplate template = UriMatchTemplate.of("/hello/{name}");

        assertTrue(template.match("/hello/John").isPresent()); // <1>
        assertEquals("/hello/John", template.expand(  // <2>
                Collections.singletonMap("name", "John")
        ));
        // end::match[]
    }
}

File: test-suite-kotlin-ksp/src/test/kotlin/io/micronaut/docs/server/uris/UriTemplateTest.kt
package io.micronaut.docs.server.uris

import io.micronaut.http.uri.UriMatchTemplate
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class UriTemplateTest {

    @Test
    fun testUriTemplate() {
        // tag::match[]
        val template = UriMatchTemplate.of("/hello/{name}")

        assertTrue(template.match("/hello/John").isPresent) // <1>
        assertEquals("/hello/John", template.expand(mapOf("name" to "John")))  // <2>
        // end::match[]
    }
}
